broker/src/main/java/io/zeebe/broker/system/SystemContext.java

```diff
@@ -70,7 +75,13 @@ private void validateConfiguration() {
       throw new IllegalArgumentException(String.format(NODE_ID_ERROR_MSG, nodeId, clusterSize));
     }
 
+    final StorageLevel storageLevel = data.getAtomixStorageLevel();
     final int replicationFactor = cluster.getReplicationFactor();
+
+    if (storageLevel == StorageLevel.MAPPED && replicationFactor > 1) {
+      throw new IllegalStateException(MMAP_REPLICATION_ERROR_MSG);
```

Comment:
this behavior is different from what is specified in the issue

> If the user sets this option to true but also uses replication we will log a warning and disable this option to not prevent the broker from starting.

Please align the two, either by adding a comment to the issue/changing the issue; or changing this here